tests/ctl/core/test_system.py

```diff
@@ -7,44 +7,47 @@
 from py._path.local import LocalPath
 
 from fides.ctl.connectors.models import OktaConfig
-from fides.ctl.core import api
 from fides.ctl.core import system as _system
+from fides.ctl.core.api import generate_resource_url
 from fides.ctl.core.config import FidesConfig
 
 
-def create_server_systems(test_config: FidesConfig, systems: List[System]) -> None:
+def create_server_systems(
+    test_config: FidesConfig, systems: List[System], test_client
+) -> None:
     for system in systems:
-        api.create(
-            url=test_config.cli.server_url,
-            resource_type="system",
-            json_resource=system.json(exclude_none=True),
+        test_client.post(
+            generate_resource_url(test_config.cli.server_url, "system"),
             headers=test_config.user.request_headers,
+            data=system.json(exclude_none=True),
         )
 
 
-def delete_server_systems(test_config: FidesConfig, systems: List[System]) -> None:
+def delete_server_systems(
+    test_config: FidesConfig, systems: List[System], test_client
+) -> None:
     for system in systems:
-        api.delete(
-            url=test_config.cli.server_url,
-            resource_type="system",
-            resource_id=system.fides_key,
+        test_client.delete(
+            generate_resource_url(
+                test_config.cli.server_url, "system", resource_id=system.fides_key
+            ),
             headers=test_config.user.request_headers,
         )
 
 
 @pytest.fixture(scope="function")
 def create_test_server_systems(
-    test_config: FidesConfig, redshift_systems: List[System]
+    test_config: FidesConfig, redshift_systems: List[System], test_client
 ) -> Generator:
     systems = redshift_systems
-    delete_server_systems(test_config, systems)
-    create_server_systems(test_config, systems)
+    delete_server_systems(test_config, systems, test_client)
+    create_server_systems(test_config, systems, test_client)
     yield
-    delete_server_systems(test_config, systems)
+    delete_server_systems(test_config, systems, test_client)
```

Comment:
This fixture is an example using the`test_client` directly which is aware of test_mode, instead of using the `api.create`/`api.delete` methods so I believe this fixture is creating the resources on the test db.  I made similar changes across several tests.

I've updated the CTL API to inject sessions into the routes, which seems to be working well.  But I think separate methods in src/fides/ctl/core/api.py are still talking to a webserver that's using the application db.   I'm not sure how to resolve yet -  will look more tomorrow - The test `test_get_all_server_systems` is failing because it's probably looking for the resources in the application db because it's calling methods in core/api.  There are a couple of other files that aren't failing per se, but are also writing to the application db.

Do you have any pointers @ThomasLaPiana?

Reply:
Thanks for weighing in, I also shared that sentiment when looking at the CLI code and helpers. First I'd like to look into untangling the CLI from the API, perhaps having both the CLI and API call the same methods instead of the CLI calling the API directly.

Reply:
I might've slightly misunderstood, but to clarify the delineation should be:

* CLI - always uses API calls to talk with the server
* API - DOES NOT use CLI-style API calls to talk to itself, instead uses its own internal functions to communicate with the database directly when needed

Reply:
I misunderstood you when you said 

> I think refactoring/removing those calls might be viable
> This is a weird (and in my opinion not optimal place) where the CLI code/helpers feel like they're a bit too intertwined with the API code.  

So I explored replacing the API calls in `fides/ctl/core/api.py` with internal functions that communicated with the database directly (that perhaps the API itself could share too, flipping what is what calling what) but I abandoned that after awhile, thinking that even though I didn't like the server calling itself, now wasn't the time to make that refactor. I didn't realize the CLI should be intentionally using API calls to talk with the server, but either way, the CLI is still using API calls to talk to the server.

I spent the remainder of the day working on your other suggestion, trying to create a client, similar to how we do in testing, instead of using the `requests` methods in `fides/ctl/core/api.py`.  Specifically I was trying to use the `httpx` `AsyncClient`.  If I pass in the app to that AsyncClient in testing I get the
RunTimeError: got Future <Future pending> attached to a different loop, I assume the client's trying to create its own loop? If I don't pass in the app, then it makes the request as expected, but against the webserver that's running the application db not the test db.

In summary, I think most of this is working (using a `test_client` is now causing it to use the test db) except when the server tries to call itself. Everything I've tried there still has it querying the application db not the test db in testing.

Reply:
To make sure we're on the same page, can you give code examples of:

CLI - always uses API calls to talk with the server. 
API - DOES NOT use CLI-style API calls to talk to itself


I want to make sure where we're talking about `cli/commands/core.py` for example versus `ctl/core/api.py` versus `ctl/routes/crud.py`

Reply:
Update: for the methods in `fides/ctl/core/api.py`, the latest commit is passing in a TestClient for test mode only which is writing to the test db .  Otherwise, it's still using `requests` library to talk to the running webserver and the application db.

Reply:
Sorry, let me clarify! Although I think you got it :) 

I don't think the webserver code logic should ever be using `fides.ctl.core.api` functions to call itself...that seems strange and is what is leading to this issue I think?

If code is running on the server, it seems like it should be able to call those functions directly instead of making an API call.

If we managed to solve this without needing to refactor all of that, I'm totally for it! There's a code smell there for sure though, at least to my "code nose" haha

Reply:
I'm still not understanding you then - can we chat tomorrow? 

The CLI is the primary item making API calls to talk to the server, and you say above that we should keep it this way because 
> There is no guarantee that the CLI will be running in the same place as the server

But here you say: 
> If code is running on the server, it seems like it should be able to call those functions directly instead of making an API call.

Reply:
@pattisdr sure! We can add it to our 1:1 agenda?

To try to sum it up in the mean time:
* The CLI should always use API calls to interact with any server logic
* The server should always call internal functions directly, i.e. if the server needs to reset the database it shouldn't call the `admin` endpoint, it should directly call the `reset_database` function

At least that's how I see it in my head....not saying it's the only way! Just feels like it has the right level of separation and maintainability

Reply:
OK great let's talk in our 1:1.  This ^ is helpful, it's not a problem to refactor things like the reset calls, but those aren't the core testing problem here, it's the CLI specifically.